File: Assets/Scripts/Falling plateform.js
#pragma strict

var plateform : GameObject;
var sound : AudioClip;
private var hasPlayed = false;


function OnTriggerEnter2D (other : Collider2D) {
    if(other.tag == "Player"){
        //Debug.Log("dected");

        if(hasPlayed == false){
            hasPlayed = true;
            GetComponent.<AudioSource>().PlayOneShot(sound);
        }

        plateform.GetComponent.<Rigidbody2D>().isKinematic = false;
    }
}

File: Assets/Scripts/FallingPlateform.cs
using UnityEngine;
using System.Collections;

public class FallingPlateform : MonoBehaviour
{
    public AudioClip sound;
    private bool hasFall = false;
    
    void OnTriggerEnter2D(Collider2D other)
    {
        if (other.tag == "Player")
        {
            if (hasFall == false)
            {
                hasFall = true;
                GetComponent<AudioSource>().PlayOneShot(sound);
                gameObject.GetComponent<Rigidbody2D>().isKinematic = false;
            }

        }
    }
}
